packages/server/lib/plugins/child/run_plugins.js

```diff
@@ -34,6 +25,14 @@ const load = (ipc, config, pluginsFile) => {
   // we track the register calls and then send them all at once
   // to the parent process
   const register = (event, handler) => {
+    const { isValid, error } = validateEvent(event, handler)
+
+    if (!isValid) {
+      ipc.send('validation:error', util.serializeError(error))
```

Comment:
missing tests around the handling of this

Reply:
@brian-mann I added an e2e test to cover this. While doing that, I discovered a bug where if an async error happens early on in the plugins file, it could end up emitting `exitEarlyWithErr` earlier than we're listening for `exitEarlyWithErr`, so it hangs the run. I fixed that with a bit of refactoring in and around `modes/run.js`, so please take a look at that too.